### PR TITLE
Bring back update download progress

### LINKED ISSUE #5408: Update Editor progress bar doesn't work on launch screen
**Describe the bug (REQUIRED)**
Update Editor progress bar doesn't work on launch screen
image

**To Reproduce (REQUIRED)**
Tap update Editor in launch screen

**Expected behavior (REQUIRED)**
It should work

**Defold version (REQUIRED):**
1.2.177

**Platforms (REQUIRED):**
 - Platforms:  macOS
 - OS: MacOs Catalina

### PR DESCRIPTION
**Pull request by vlaaad**
Now the update download progress is back

Fixes #5408

### CODE REVIEW

File: editor/src/clj/util/net.clj [hunk 1]
```diff
@@ -34,31 +31,30 @@
...
-                        (.write output buf 0 count)
-                        (progress-callback (+ previous count) length)
-                        (recur (.read input buf) (+ previous count)))))
-                  (do
-                    ;; io/copy will use an internal buffer of 1024 bytes for transfer
-                    (io/copy input output)
-                    (when (not @cancelled-derefable)
-                      (progress-callback length length)))))))))
+      301 (let [location (.getHeaderField conn "Location")]
+            (apply download! location out (mapcat identity args)))
```
Comment (vlaaad) [inline editor/src/clj/util/net.clj:44]: Here is the actual fix: from `(download! location out)` to `(apply download! location out (mapcat identity args))` so we don't lose progress tracker on redirects.

**Comment by vlaaad** [comment]
I tested this stuff locally

Seal of approval

**Comment by AGulev** [comment]
You have removed PR Checklist and didn't add `skip release notes`, so I've added it for you.
Please, pay attention to `PR Checklist` or add `skip release notes` if it's not something you want in Release Notes.
Release process takes much longer, if we don't pay attention to it

**Comment by vlaaad** [comment]
@AGulev I've read the checklist and didn't assign the label to this PR because the checklist says to assign it on the affected issue — there is none.

**Comment by AGulev** [comment]
> @AGulev I've read the checklist and didn't assign the label to this PR because the checklist says to assign it on the affected issue — there is none.


https://github.com/defold/defold/issues/5408

**Comment by vlaaad** [comment]
@AGulev, thanks!

**Review by matgis — APPROVED** [review]
Ship it!

**Comment by aglitchman** [comment]
Finally! Many thanks! ❤️

**Pull request merged**